serving/purgatorio/select-and-train-machine-learning-models/tools-and-libraries/tools-and-libraries.md

```diff
@@ -1,5 +1,321 @@
+
+For a comprehensive list of production-ready tools, be sure to take a tour at:
+
+- [Awesome Production Machine Learning](https://github.com/EthicalML/awesome-production-machine-learning#data-labelling-tools-and-frameworks)
+
+Virgilio has chosen for you the best tools of this type for 3 problem categories:
+
+## Computer Vision
+
+- [OpenCV](https://opencv.org/about/)
+
+OpenCV (Open Source Computer Vision Library) is an open-source computer vision and machine learning software library. OpenCV was built to provide a common infrastructure for computer vision applications and to accelerate the use of machine perception in commercial products. Being a BSD-licensed product, OpenCV makes it easy for businesses to utilize and modify the code.
+
+Learn how to use OpenCV here:
+
+- [OpenCV Python for Beginners - Full Course in 10 Hours (2020)](https://www.youtube.com/watch?v=N81PCpADwKQ)
+
+Check also Virgilio's tutorials in the Inferno section, They all use OpenCV!
+
+## Natural Language Processing
+
+- [Spacy](https://spacy.io/)
+
+SpaCy is a free, open-source library for advanced Natural Language Processing (NLP) in Python.
+
+If you’re working with a lot of text, you’ll eventually want to know more about it. For example, what’s it about? What do the words mean in context? Who is doing what to whom? What companies and products are mentioned? Which texts are similar to each other?
+
+spaCy is designed specifically for production use and helps you build applications that process and “understand” large volumes of text. It can be used to build information extraction or natural language understanding systems or to pre-process text for deep learning.
+
+They offer a free course on their website:
+
+- [Advanced NLP with Spacy](https://course.spacy.io/)
+
+
+## Time Series Forecasting
+
+- [Prophet](https://facebook.github.io/prophet/)
+
+Prophet is a procedure for forecasting time series data based on an additive model where non-linear trends are fit with yearly, weekly, and daily seasonality, plus holiday effects. It works best with time series that have strong seasonal effects and several seasons of historical data. Prophet is robust to missing data and shifts in the trend, and typically handles outliers well.
+
+Prophet is open source software released by Facebook’s Core Data Science team. It is available for download on CRAN and PyPI.
+
+Get the tutorials here:
+
+- [Prophet Quick Start](https://facebook.github.io/prophet/docs/quick_start.html)
+- [A Guide to Time Series Forecasting with Prophet in Python 3](https://www.digitalocean.com/community/tutorials/a-guide-to-time-series-forecasting-with-prophet-in-python-3)
+- [Forecasting in Python with Facebook Prophet](https://towardsdatascience.com/forecasting-in-python-with-facebook-prophet-29810eb57e66)
+
```

Comment:
It's a great set of resources above, pls have a look at this one too https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/time-series_anomaly-detection/README.md#time-series--anomaly-detection

Maybe mention it for further reads.

Reply:
added!

Reply:
Actually they don't mention or use prophet, so they don't fit here unluckily

Reply:
Prophet is not the only timeseries, library although it's widely used and if you do take a look at https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/time-series_anomaly-detection/README.md#time-series--anomaly-detection, prophet is mentioned, please search for "Prophet" on the page you will see multiple references.

One should know other things in the world of timeseries as well as the main topic is Timeseries.

Reply:
sure, but the point here is to choose one among the dozens available.. it's easy for everyone google "time series libraries" but it's not the point of this section of the guide :)